### DISCUSSION TITLE
Public users registration with roles

### DISCUSSION THREAD

**Discussion by alcompstudio** [directus/directus#16977] [Feature Requests]
Hello! I am not a programmer, so I would like to see the implementation, in my opinion, of a very useful functionality - user registration in the Project.
That is, the user, following the link, for example https://site.directus.app/admin/login - as usual, gets to the authorization page. But it can also register, without the participation of an administrator, while receiving one of the so-called "public" roles. 
![image](https://user-images.githubusercontent.com/22731880/210153118-5458a854-8ea3-4de0-bf09-5ad102a12857.png)

The display of a public role in the registration form is configured in the settings.
![image](https://user-images.githubusercontent.com/22731880/210153207-eae30e0b-c8cc-46a2-82d7-a9b428eb02e9.png)

Further, after registration, the user already independently configures his user page fields (those that he is allowed to configure, of course).

You can also develop a user activation script - the administrator manually activates a new user, or automatic activation occurs, after which the new authorized (and activated) user can work in the system according to his configured access rights.

I want to make a portal for Customers and Contractors, where each user chooses a role for himself and already uses the configured functionality inside the application.

I'm just learning this web service, I really like its concept, but the obvious question immediately arose about public user registration.
Maybe it's already implemented somehow and I just can't find where to enable this functionality? If not, can Directus developers implement it? How to find out?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
